glom/core.py

```diff
@@ -621,6 +621,10 @@ def __init__(self, *path_parts):
                 while i < len(sub_parts):
                     path_t = _t_child(path_t, sub_parts[i], sub_parts[i + 1])
                     i += 2
+            elif part == '*':
+                path_t = _t_child(path_t, 'x', None)
+            elif part == '**':
```

Comment:
Ah, so `Path('*')` would also traverse. To me, this seems to turn `'*'` into a maybe-overaccessible singleton. 

What if we did this in `Path.from_text()`? Then `'a.*.b` would still work, but we replace `'*'` segments with a new `STAR` singleton (or `T.__star__`). Then `Path` remains faithful to the segment-preserving behavior. After all, `Path` was created for disambiguating between `'1'` and `1`, as well as segments with dots in them.

Reply:
Thinking about this for a bit, there's an invariant that `Path('a', 'b') == Path.from_text('a.b')` so, similarly `Path('a', '*') == Path.from_text('a.*')`

I see your point that `Path('a', 1) != Path.from_text('a.1')` so there's precedent.  Mechanically I think you are right, having `from_text()` transform segments to `T.__star__()` or `T.__starstar()` would implement the behavior.

Reply:
I think of `Path` as kind of the coercing, convenient operator and `T` as the precision tool.

`Path()` being more precise than `Path.from_text` seems kind of incidental.

Another idea I had early on was `MultiPath` -- so `MultiPath('a.*.b')` might be unambiguous.

The main think I think is that `Auto('**')` kind of has the new behavior

Reply:
Great :) As an aside, i don't think we need `.__star__()` to be called like that, it could be just `.__star__` to start maybe?

Reply:
I hear you -- I'm kinda wondering if the API should even be `.__('*')` since `*` isn't a valid python glyph

`__star__` is better than `__star__()` because it is shorter; but maybe `__('*')` is even better